include/deal.II/sundials/n_vector.templates.h

```diff
@@ -170,6 +171,86 @@ namespace SUNDIALS
 #  endif
     );
 
+
+    template <typename VectorType>
+    constexpr bool is_dealii_compatible_distributed_vector =
+      std::is_same_v<
+        VectorType,
+        LinearAlgebra::distributed::Vector<typename VectorType::value_type,
+                                           MemorySpace::Host>> ||
+      std::is_same_v<VectorType,
+                     LinearAlgebra::distributed::BlockVector<
+                       typename VectorType::value_type>>;
+
+    template <typename VectorType,
+              std::enable_if_t<!IsBlockVector<VectorType>::value, VectorType>
```

Comment:
You don't actually care what the specific pointer type is. Might as well make it a `void*`:
```suggestion
              std::enable_if_t<!IsBlockVector<VectorType>::value>
```
Same below?

Reply:
Changed it in all places.